### PR TITLE
[DD4hep] start on geometry XML payload producer

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

@cvuosalo - IMHO, this is a very straightforward way: if `xml_geometry_payload` is defined, a big XML will be produced (`std::cout` for now).

Please, see the test:
```
cmsRun DetectorDescription/DDCMS/test/python/testShapes.py
``` 
 
#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
@cvuosalo - please, see https://github.com/cms-sw/cmssw/pull/33548/commits/5539cc7cb391c6e226d4256a04aa1322cf3c3939

**Comment by perrotta** [comment]
-1
(please rebase in order to get rid of unwanted commits)

File: RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEFastESProducer.cc [hunk 1]
```diff
@@ -92,6 +92,8 @@ void PixelCPEFastESProducer::fillDescriptions(edm::ConfigurationDescriptions& de
   // specific to PixelCPEFastESProducer
   desc.add<std::string>("ComponentName", "PixelCPEFast");
   desc.add<edm::ESInputTag>("MagneticFieldRecord", edm::ESInputTag());
+  desc.addOptional<bool>("useLAAlignmentOffsets", false)->setComment("deprecated");
```
Comment (mmusich) [inline RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEFastESProducer.cc:95]: @ianna why this change?

File: RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEGenericESProducer.cc [hunk 2]
```diff
@@ -105,6 +105,8 @@ void PixelCPEGenericESProducer::fillDescriptions(edm::ConfigurationDescriptions&
   // specific to PixelCPEGenericESProducer
   desc.add<std::string>("ComponentName", "PixelCPEGeneric");
   desc.add<edm::ESInputTag>("MagneticFieldRecord", edm::ESInputTag(""));
+  desc.addOptional<bool>("useLAAlignmentOffsets", false)->setComment("deprecated");
```
Comment (mmusich) [inline RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEGenericESProducer.cc:108]: @ianna why this change?

File: RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEFastESProducer.cc (hunk 1 above)
Comment (ianna) [reply RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEFastESProducer.cc:95]: I think rebase didn't take into an account reverted changes :-(

File: RecoLocalTracker/SiPixelRecHits/plugins/PixelCPETemplateRecoESProducer.cc [hunk 3]
```diff
@@ -83,6 +83,7 @@ void PixelCPETemplateRecoESProducer::fillDescriptions(edm::ConfigurationDescript
 
   // specific to PixelCPETemplateRecoESProducer
   desc.add<std::string>("ComponentName", "PixelCPETemplateReco");
+  desc.addOptional<bool>("DoLorentz", true)->setComment("deprecated");
```
Comment (mmusich) [inline RecoLocalTracker/SiPixelRecHits/plugins/PixelCPETemplateRecoESProducer.cc:86]: @ianna why this change?

File: RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEGenericESProducer.cc (hunk 2 above)
Comment (ianna) [reply RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEGenericESProducer.cc:108]: I think rebase didn't take into an account reverted changes:
```

Revert "Update mtdParameters for D49 and D60 (Bug fix)", This reverts…  … 1307d75

Revert "[backport] MTD geometry and reconstruction: synchronise with …  … faf18ac

change ECAL pedestals requireStableBeam flag for MWGR test        2a9cbcd

Revert "Revert "[backport] MTD geometry and reconstruction: synchroni…  …f2a994f

revert changes breaking online HLT menu in production release         93afbed
```

**Comment by fabiocos** [comment]
@ianna why restoring old parameters in ```mtdParameters.xml```?

**Comment by fabiocos** [comment]
I guess it is a similar issue as in the discussion with @mmusich ... There is no reason to go back anyway

**Comment by ianna** [comment]
> I guess it is a similar issue as in the discussion with @mmusich ... There is no reason to go back anyway

Yes, I have no idea why a rebase picked the other commits. I'll try to rebase again a bit later. If this will not work, I'll cherry-pick my commit and open a new PR.

**Comment by civanch** [comment]
@ianna , better to start a new PR even if extra work will be needed.

**Comment by slava77** [comment]
-1

it looks like the rebase went wrong and changes in reco code are not expected

**Comment by srimanob** [comment]
@ianna 
Do we still need this, or should it be closed? Thanks.
